test/e2e_node/node_perf_test.go

```diff
@@ -126,12 +126,29 @@ var _ = SIGDescribe("Node Performance Testing [Serial] [Slow]", func() {
 		// Create the pod.
 		pod = f.PodClient().CreateSync(pod)
 		// Wait for pod success.
-		f.PodClient().WaitForSuccess(pod.Name, wl.Timeout())
+		// but avoid using WaitForSuccess because we want the container logs upon failure #109295
+		podErr := e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, pod.Name, fmt.Sprintf("%s or %s", v1.PodSucceeded, v1.PodFailed), wl.Timeout(),
+			func(pod *v1.Pod) (bool, error) {
+				switch pod.Status.Phase {
+				case v1.PodFailed:
+					return true, fmt.Errorf("pod %q failed with reason: %q, message: %q", pod.Name, pod.Status.Reason, pod.Status.Message)
+				case v1.PodSucceeded:
+					return true, nil
+				default:
+					return false, nil
+				}
+			},
+		)
 		podLogs, err := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, pod.Spec.Containers[0].Name)
 		framework.ExpectNoError(err)
+		if podErr != nil {
+			framework.Logf("dumping pod logs due to pod error detected:%s", podLogs)
+		}
 		perf, err := wl.ExtractPerformanceFromLogs(podLogs)
 		framework.ExpectNoError(err)
 		framework.Logf("Time to complete workload %s: %v", wl.Name(), perf)
+		// using framework.ExpectNoError for consistency would cause changes the output format
+		gomega.Expect(podErr).To(gomega.Succeed(), "wait for pod %q to succeed", pod.Name)
```

Comment:
you already check that pofErr is != nil, seems redundant to check if it succeed here

Reply:
https://github.com/kubernetes/kubernetes/pull/111015/files#r917446567

Reply:
I am going through some effort here to preserve the output formatting of the existing test case because somebody may have some automation on these test results.  That's why I put that comment on line 150.  I could remove the "!= nil" check and dump the logs every time, but that might be a problem since these are long-running tests and may generate a lot of output.

The "podErr != nil" check on line 142 seems minimally intrusive, preserves the expected format of the test output, and keeps it looking like a normal test case. I'm no gomega expert. Is there an alternative that is easy to read at a glance and preserves the functionality?

Reply:
> I am going through some effort here to preserve the output formatting of the existing test case because somebody may have some automation on these test results.

the shouldn't ;)

> ut that might be a problem since these are long-running tests and may generate a lot of output.

yeah

> The "podErr != nil" check on line 142 seems minimally intrusive, preserves the expected format of the test output, and keeps it looking like a normal test case. I'm no gomega expert. Is there an alternative that is easy to read at a glance and preserves the functionality?

this lgtm, @endocrimes for approval